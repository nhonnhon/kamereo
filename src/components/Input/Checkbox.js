import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './Checkbox.scss'

class Checkbox extends Component {
  static propTypes = {
    checked: PropTypes.bool,
    blocked: PropTypes.bool,
    children: PropTypes.node,
    text: PropTypes.string,
    center: PropTypes.bool,
    className: PropTypes.string,
    id: PropTypes.string,
    color: PropTypes.string,
    onCheck: PropTypes.func,
  }

  static defaultProps = {
    color: '',
    blocked: false,
  }

  toggle = e => {
    e.stopPropagation()
    this.props.onCheck()
  }

  render() {
    const { children, text, center, className, id, color, blocked, checked } = this.props
    const statusHold = checked ? 'checked' : ''
    const status = blocked ? 'blocked' : statusHold
    return (
      <div
        className={`checkbox ${status} ${className || ''}`}
        style={{ alignItems: center ? 'center' : '' }}
        onClick={this.toggle}
      >
        <div className="icon" />
        <input className="hide" id={id} type="checkbox" />
        {text ? (
          <label className={`pad-left ${color}`} htmlFor={id}>
            {text}
          </label>
        ) : (
          <label className={`pad-left ${color}`} htmlFor={id}>
            <div className={`content`}>{children}</div>
          </label>
        )}
      </div>
    )
  }
}

export default Checkbox
