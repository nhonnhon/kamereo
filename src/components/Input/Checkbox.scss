@import '../../assets/css/base.scss';

.checkbox {
  display: block;
  min-height: 20px;
  position: relative;
  cursor: pointer;
  > .icon {
    height: 20px;
    position: absolute;
    width: 20px;
    border: 1px solid $gray1;
    background-color: $white;
  }
  &.checked {
    > .icon {
      background-color: $green;
    }
  }
  > label {
    display: inline-block;
    font-weight: 300;
    line-height: 1.57em;
    &.white {
      color: #ffffff;
      span,
      p {
        color: #ffffff;
      }
    }
  }
}
