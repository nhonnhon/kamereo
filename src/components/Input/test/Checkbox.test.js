import React from 'react'
import { create } from 'react-test-renderer'
import Checkbox from '../Checkbox'

describe('Button component', () => {
  test('Matches the snapshot', () => {
    const button = create(<Checkbox />)
    expect(button.toJSON()).toMatchSnapshot()
  })
})
