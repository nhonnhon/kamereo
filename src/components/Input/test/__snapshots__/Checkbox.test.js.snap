// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Matches the snapshot 1`] = `
<div
  className="checkbox  "
  onClick={[Function]}
  style={
    Object {
      "alignItems": "",
    }
  }
>
  <div
    className="icon"
  />
  <input
    className="hide"
    type="checkbox"
  />
  <label
    className="pad-left "
  >
    <div
      className="content"
    />
  </label>
</div>
`;
